### ISSUE TITLE
Problems with git (and composer)

### ISSUE THREAD

**Issue by HnLn** [GM-Alex/vagrant-winnfsd#22]
I have a vagrant box (configured with puphpet) and when running git clone command from within the virtualbox I get errors that certain files can't be deleted.  I had a similar issue with jankowfsky winnfsd: https://bitbucket.org/jankowfsky/winnfsd/issue/9/create-acces-ok-write-fail.

I've researched a bit further as I started to realise that it seems the problem only occurs with .git folders (also with composer that uses git).  Below is a log from proces monitor as I suspected that maybe the windows host was trying to access the files as well (sourctree or git bash).  According the log, that doesn't seem to be the case.  Problem seems to be with SetDispositionInformationFile (no idea what that is / does).

The log, the 5th entry is the one causing troubles.
```
"Time of Day","Process Name","PID","Operation","Path","Result","Detail"
"13:08:21.0964796","winnfsd.exe","7544","CloseFile","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects","SUCCESS",""
"13:08:21.0964898","winnfsd.exe","7544","IRP_MJ_CLOSE","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects","SUCCESS",""
"13:08:21.0965313","winnfsd.exe","7544","CreateFile","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects\pack\tmp_idx_G0HItM","SUCCESS","Desired Access: Read Attributes, Delete, Disposition: Open, Options: Non-Directory File, Open Reparse Point, Attributes: n/a, ShareMode: Read, Write, Delete, AllocationSize: n/a, OpenResult: Opened"
"13:08:21.0968146","winnfsd.exe","7544","QueryAttributeTagFile","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects\pack\tmp_idx_G0HItM","SUCCESS","Attributes: RANCI, ReparseTag: 0x0"
"13:08:21.0968269","winnfsd.exe","7544","SetDispositionInformationFile","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects\pack\tmp_idx_G0HItM","CANNOT DELETE","Delete: True"
"13:08:21.0968839","winnfsd.exe","7544","CloseFile","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects\pack\tmp_idx_G0HItM","SUCCESS",""
"13:08:21.0969287","winnfsd.exe","7544","IRP_MJ_CLOSE","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects\pack\tmp_idx_G0HItM","SUCCESS",""
"13:08:21.0969780","winnfsd.exe","7544","CreateFile","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects","SUCCESS","Desired Access: Read Data/List Directory, Synchronize, Disposition: Open, Options: Directory, Synchronous IO Non-Alert, Attributes: n/a, ShareMode: Read, Write, Delete, AllocationSize: n/a, OpenResult: Opened"
"13:08:21.0970129","winnfsd.exe","7544","QueryDirectory","D:\www\culturefeed_instance\vendor\symfony\console\Symfony\Component\Console\.git\objects\pack","SUCCESS","Filter: pack, 1: pack"
```

**Comment by michael-franz** [comment]
It seems the problem is gone with version 1.0.11.

**Comment by worthwhileindustries** [comment]
what version/repo of WinNFSd is currently being used in the project?

**Comment by GM-Alex** [comment]
Yann is part of the winnfsd project (https://bitbucket.org/jankowfsky/winnfsd), which also part of this project. All of his changes are merged with the latest version and also included in the latest version of the module.

**Comment by eXtreme** [comment]
With 1.0.11 many problems have been solved by i'm suffering from some new as well - can live with them but they are annoying. Anyway, it's great to see this project!

1. with composer update/install I get an error that my `composer.lock` or `vendor/composer/installed.json` is corrupted (just after packages are updated) - at the end of these files there are some *leftovers* from previous version of the file? Json is corrupted with some old lines. I manually remove them and run the command again and it works.
2. sometimes files are not updated on guest after modifying on host. I have to wait longer or make more more changes or even change other files to "trigger" updating. 
3. sometimes new files in a folder are not transferred to guest. Reproducible situation on my machine: create a folder, create some files there, modify them, then create new files in that folder - new files are never transferred to guest. I have to `vagrant reload` to make them appear.

If there is a way I can provide you some logs or anything - **tell me how**, I will reproduce these problems and create new issue.

**Comment by GM-Alex** [comment]
All these issue are related to the winnfsd project (https://bitbucket.org/jankowfsky/winnfsd), so it would be better to continue the discussion there. I will close the issue here.

**Comment by lukebennett** [comment]
Just for the record, I'm unable to git pull/commit/stash etc (basically anything that manipulates the .git directory) in 1.0.11, including with Yann's forked executable. Unfortunately it would appear that any fix has stalled as there's been no activity on the winnfsd issue for a couple of months.

**Comment by GM-Alex** [comment]
Yes, Yann and I are full with work to do, so there is no time at the moment for that. Maybe someone with some C++ experience want's to join. We are using git directly on windows at my company what works without any problems, that's no solution but a good workaround for the daily work.

**Comment by lukebennett** [comment]
Sure, would help if I could but sadly I don't do C++. We're able to use Git on Windows fine, we just need the ability to do so within the Vagrant guest as well. We'll sit tight and wait on this I guess as none of the other Vagrant synced folder implementations are really suitable.

**Comment by GM-Alex** [comment]
You could test http://www.hanewin.de/nfs-d.htm maybe that one will work with git. If it's like that it could be a option to extend the vagrant module to check if some installed the alternative server (which will be easier than fixing the issue), but this one is shareware.

**Comment by lukebennett** [comment]
Actually, it looks like I may have goofed up here - despite installing the 1.0.11 plugin, it turns out Vagrant was still using 1.0.10 for some reason. Having forcefully removed that and made sure 1.0.11 is in use, I'm not getting the issue any more.

**Comment by worthwhileindustries** [comment]
@lukebennett how did you find that out? I noticed that vagrant keeps old plugins and often wonder if it is using an old one? Did you just do a vagrant plugin list and it showed the old version?

**Comment by lukebennett** [comment]
@worthwhileindustries Pure luck really - at no point did I explicitly upgrade any of my plugins. I started noticing some adverse behaviour with a different plugin and after investigating realised Vagrant had automatically installed new versions for both that and winnfsd.

As the other plugin was definitely running on the new version, I assumed winnfsd was too, Vagrant seemingly having updated both without prompting. What made me realise winnfsd was still on the old version was when I happened to spot the old exe running in Task Manager.

If anybody knows how Vagrant decides when to install and/or switch plugins, feel free to enlighten me :)

**Comment by Jeremlicious** [comment]
Hi, I'm using Vagrant 1.7.2 and vagrant-winfsd 1.0.11 on a Windows 8.1 and I'm still getting weird issues like "Input/output error"


Example on composer install : 
```
 Could not delete /dev/project/vendor/name/lib/.git/objects/pack/tmp_idx_SeJiQh:
```

**Comment by krisdante** [comment]
Same for me. I'll try to use Vagrant started as Administrator today, and it that also fail I'll just give up with NFS.

**Comment by Jeremlicious** [comment]
I found a workaround concerning vendor installation. Use dist package so there wont be a .git folder where files can't be deleted

` php composer.phar install --prefer-dist`

**Comment by razbakov** [comment]
I am using 1.0.11, i tried workaround jankowfsky/winnfsd and `php composer.phar install --prefer-dist`.
Nothing works for me

**Comment by LukasVice** [comment]
:+1:

**Comment by mreichenbach-igniti** [comment]
I have the same problem like @Jeremlicious with the same versions of Vagrant and vagrant-winnfsd, but only using Windows 7. I also checked the version of `winnfsd.exe` which is the current v2.0

**Comment by luciansabo** [comment]
I personally switched to Linux because of this issue

**Comment by worthwhileindustries** [comment]
I am using global install of composer installed on a non-shared directory within the vm. I use composer within the vm... not on windows. Maybe this is the problem? I've never ever had a single issue using composer to update my shared libraries in my git repo in the shared directory and I've been using this plugin for 6+ months everyday. I'm currently running Windows 8.1. VirtualBox 4.3.22

**Comment by TheFamousPG** [comment]
:+1:

**Comment by MetalArend** [comment]
Where exactly on the winnfsd issues tracker do they follow up on this issue? @GM-Alex, could you please reference some specific issue, if there is one? I keep running into this, and would love to help to get some insights if it would help to solve the problem.

**Comment by eXtreme** [comment]
@MetalArend https://bitbucket.org/jankowfsky/winnfsd/issues/7/winnfsd-doesnt-always-overwrite-files this issue provides latest compiled .exes with newest fixes.

**Comment by johnnyborg** [comment]
@eXtreme you can read it here; https://bitbucket.org/jankowfsky/winnfsd/issues/9/create-acces-ok-write-fail. You can replace the current binaries provided by Yann Schepens to resolve this ticket. I don't when @GM-Alex will provide a new version of the Vagrant plugin.

**Comment by yannschepens** [comment]
@Johnnyborg: Thk
Le ven. 10 juil. 2015 à 10:54, johnnyborg <notifications@github.com> a
écrit :
